### PR TITLE
Add multiple work types

### LINKED ISSUE #979: Generate all work types
~* Change Work to Generic Work~
* Generate:
  * Article
  * Image
  * Document
  * Student Work
  * ETD
  * Dataset
  * Video

For now, we'll stick with default metadata - will add all the fields and predicates after metadata SME review.

### PR DESCRIPTION
**Pull request by crowesn**
Fixes #979 

Add subsequent work-types with shared controller logic and specs.

Changes proposed in this pull request:
* Add work types
* Add concern with shared controller logic
* Fill out shared example specs

### CODE REVIEW

File: app/controllers/concerns/scholar/works_controller_behavior.rb [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+      super
+      curation_concern.read_groups = ["public"]
+    end
+
+    def create
+      super
+    rescue ActiveFedora::RecordInvalid # virus detected
+      remove_infected_file_sets
+      report_virus_found
+      redirect_to main_app.curation_concerns_generic_work_path(id: curation_concern.id)
```
Comment (crowesn) [inline app/controllers/concerns/scholar/works_controller_behavior.rb]: Eh, we need polymorphic here. I'll work on it.

**Comment by jamesvanmil** [comment]
I have some thoughts about testing - the multiple work types aren't being tested with the `work_show_spec` or the `virus_scan_spec`, which I think are the only two work-centric feature tests that don't test the additional work types.

I don't think it's important to test each work type with these additional feature spec. The `work_show_spec` is largely duplicated by the `create_work_spec` and the `end_to_end_spec`. The `virus_scan_spec` is testing code that is largely in the controllers, and the feature test was an alternative to testing on the controller level.

Adding examples for each of these wouldn't scale well because of added time in the test suite, and other problems we have with some of these tests and application responsiveness, but skipping them leaves a significant coverage gap for virus scanning.

Should we take another look at testing the virus scanning functionality in controller specs so we can test all work types?

**Comment by crowesn** [comment]
Yes

**Comment by hortongn** [comment]
👍 to no need for virus feature tests for each work type

**Pull request merged**